frontend/src/lib/components/ProductEmptyState/ProductEmptyState.tsx

```diff
@@ -0,0 +1,72 @@
+import { LemonButton } from 'lib/lemon-ui/LemonButton'
+import { IconOpenInNew, IconPlus } from 'lib/lemon-ui/icons'
+import { BuilderHog3, DetectiveHog } from '../hedgehogs'
+
+export const ProductEmptyState = ({
+    productName,
+    thingName,
+    description,
+    action,
+    actionElementOverride,
+    docsURL,
+    customHog,
+}: {
+    // The name of the product, e.g. "Cohorts"
+    productName: string
+    // The name of the thing that they will create, e.g. "cohort"
+    thingName: string
+    description: string
+    // The action to take when the user clicks the CTA
+    action?: () => void
+    // If you want to provide a custom action button instead of using the default one
+    actionElementOverride?: JSX.Element
+    docsURL?: string
+    customHog?: JSX.Element
```

Comment:
I think this could be more obvious for the future if there's a set mapping from hedgehog name to component, in which case a developer using `ProductEmptyState` only needs to provide a hedgehog string like "builder". Fewer ways for things to go wrong, e.g. if someone doesn't set proper width/height classes. Although I can't see any usage of `customHog` right now, so maybe you have some different direction in mind.